### PR TITLE
fix(Scripts/ICC): Use threatlist for vampiric bite

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by TheSCREWEDSoftware** [comment]
@Exitare are you able to test this?

**Comment by Nyeriah** [comment]
Use the built in target selector instead Unit* target = SelectTarget(...)

It has a predicate argument

**Comment by Exitare** [comment]
Yes, I can check it. Taking a break for the holidays though. Hope that's ok.

**Comment by Nyeriah** [comment]
Not ready for testing yet need rewrite

**Comment by Kitzunu** [comment]
It should be good now

**Review by Nyeriah — APPROVED** [review]

**Comment by Exitare** [comment]
So i tested it. And i think there is an issue with the event scheduling itself. 
I am not totally mistaken, the event is being scheduled to be cast after about 15s of starting the encounter.
But its never rescheduled? Is that intended? 

I couldnt verify that it was cast at least once via log and attaching the debugger inside of the switch case. It never trggered.

File: src/server/scripts/Northrend/IcecrownCitadel/boss_blood_queen_lana_thel.cpp [hunk 1]
```diff
@@ -327,22 +338,7 @@ class boss_blood_queen_lana_thel : public CreatureScript
...
-                                    float th = me->GetThreatMgr().getThreatWithoutTemp(p);
-                                    if (!target || th > maxThreat)
-                                    {
-                                        target = p;
-                                        maxThreat = th;
-                                    }
-                                }
-
-                        if (target)
+                        if (Unit* target = SelectTarget(SelectTargetMethod::MaxThreat, 0, VampiricBiteTargetSelector(me)))
```
Comment (Exitare) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_blood_queen_lana_thel.cpp:341]: There is no scheduling in this function, compared to all other abilities of the fight. Is that correct?

**Comment by Nyeriah** [comment]
> But its never rescheduled? Is that intended?

Yes, she only bites one player, then players bite each other. It is fine.

**Comment by Exitare** [comment]
Right, I remember now. 
I tested again. It does work. IIRC it also shouldn't select the tank / highest threat, which it didn't. So that worked too.

File: src/server/scripts/Northrend/IcecrownCitadel/boss_blood_queen_lana_thel.cpp [hunk 2]
```diff
@@ -79,25 +80,10 @@ enum Shadowmourne
...
-    {70877, 71474, 70877, 71474},
-};
-
 bool IsVampire(Unit const* unit)
 {
-    uint8 spawnMode = unit->GetMap()->GetSpawnMode();
-    for (uint8 i = 0; i < 3; ++i)
-        if (unit->HasAura(vampireAuras[i][spawnMode]))
-            return true;
+    if (unit->HasAura(SPELL_ESSENCE_OF_BLOOD_QUEEN) || unit->HasAura(SPELL_ESSENCE_OF_THE_BLOOD_QUEEN_PLR) || unit->HasAura(SPELL_FRENZIED_BLOODTHIRST))
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_blood_queen_lana_thel.cpp:85]: Let me rework this to HasAnySpells before merge

Comment (Kitzunu) [inline src/server/scripts/Northrend/IcecrownCitadel/boss_blood_queen_lana_thel.cpp:85]:
```suggestion
    if (unit->HasAnyAuras(SPELL_ESSENCE_OF_BLOOD_QUEEN, SPELL_ESSENCE_OF_THE_BLOOD_QUEEN_PLR, SPELL_FRENZIED_BLOODTHIRST))
```

**Pull request merged**